gwnr/waveform/enigma_utils.py

```diff
@@ -242,10 +272,180 @@ def get_imr_enigma_modes(
+        f_lower                 -- Starting frequency of the waveform (in Hz)
+        sample_rate             -- Sampling rate of the waveform timeseries (in Hz)
+        spin1z, spin2z          -- z-components of component dimensionless spins (lies in [0,1))
+        eccentricity            -- Initial eccentricity
+        mean_anomaly            -- Mean anomaly of the periastron (in rad)
+        inclination             -- Inclination (in rad), defined as the angle between the orbital 
+                                   angular momentum L and the line-of-sight
+        coa_phase               -- Coalesence phase of the binary (in rad)
+        distance                -- Luminosity distance to the binary (in Mpc)
+        modes_to_use            -- GW modes to use. List of tuples (l, |m|)
+        return_orbital_params   -- If True, returns the orbital evolution of all the orbital elements (in 
+                                   geometrized units). Can also be a list of orbital variable names to return 
+                                   only those specific variables. Available orbital variables names are: 
+                                   ['x', 'e', 'l', 'phi', 'phidot', 'r', 'rdot']
+        verbose                 -- Verbosity flag
+    
+    Returns:
+    --------
+        t                 -- Time grid (in seconds)
+        orbital_var_dict  -- Dictionary of evolution of orbital elements. 
+                             Returned only if "return_orbital_params" is specified.
+        hp, hc            -- Plus and cross GW polarizations
+    """
+
+    retval = get_inspiral_enigma_modes(
+        mass1=mass1,
+        mass2=mass2,
+        spin1z=spin1z,
+        spin2z=spin2z,
+        eccentricity=eccentricity,
+        mean_anomaly=mean_anomaly,
+        distance=distance,
+        f_lower=f_lower,
+        sample_rate=sample_rate,
+        modes_to_use=modes_to_use,
+        include_conjugate_modes=True,  # Always include conjugate modes while generating polarizations
+        return_orbital_params=return_orbital_params,
+        verbose=verbose,
+    )
+
+    if return_orbital_params:
+        t, orbital_var_dict, retval = retval
+    else:
+        t, retval = retval
+
+    modes_imr = retval
+
+    hp_ihc = modes_imr[(2, 2)] * 0  # Initialize with zeros
     if verbose:
-        print("Modes generation took: {} seconds".format(time_delta))
+        print("Shape of hp_ihc: {}".format(np.shape(hp_ihc)), flush=True)
 
+    for el, em in modes_imr:
+        ylm = lal.SpinWeightedSphericalHarmonic(inclination, coa_phase, -2, el, em)
+        hp_ihc = hp_ihc + modes_imr[(el, em)] * ylm
+        if verbose:
+            print(f"Adding mode {el}, {em} with ylm = {ylm}", flush=True)
+            print(
+                "... adding {}, {}".format(
+                    modes_imr[(el, em)], modes_imr[(el, em)] * ylm
+                ),
+                flush=True,
+            )
```

Comment:
This will print a lot of gunk. Please hide this print behind another flag than just verbose. 

My current convention is to use `verbosity=0` flag, which if its 0 means `verbose=False` and as it becomes a larger integer, the more verbose output can be provided..